### PR TITLE
[insights-agent] 2.0 Release

### PR DESCRIPTION
**Pull request by makoscafee**
**Why This PR?**
_a short description of why this PR is needed_
This is the final PR that merges all `Agent 2.0` changes to `master `branch
Fixes #

**Changes**
Changes proposed in this pull request:

* Add Hyphen back to all reports
* Rename resource metrics to prometheus-metrics
* Change admission failurePolicy to ignore 

**Checklist:**

* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] Any new values have been added to the README for the Chart, or `helm-docs --sort-values-order=file` has been run for the charts that support it.

### CODE REVIEW

File: stable/insights-admission/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: insights-admission
 description: A Validating Webhook Admission Controller that utilizes Fairwinds Insights.
 type: application
-version: 0.4.5
+version: 0.5.0
```
Comment (rbren) [inline stable/insights-admission/Chart.yaml]: This should be 1.0.0

File: stable/insights-agent/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: A Helm chart to run the Fairwinds Insights agent
 name: insights-agent
-version: 1.17.16
+version: 1.18.0
```
Comment (rbren) [inline stable/insights-agent/Chart.yaml]: This should be `2.0.0`

**Review by rbren — COMMENTED** [review]
I think we will need to do this in two parts:
* release insights-admission 1.0.0
* release insights-agent 2.0.0 (with insights-admission 1.0.0 as a dependency)

To do that, you can create a new branch (say, `bm/admission-1-0`) and run
```
git checkout bm/agent-2-0 ./stable/insights-admission
```

**Review by rbren — COMMENTED** [review]
We also need to update polaris to 5.0 right?

**Comment by makoscafee** [comment]
> I think we will need to do this in two parts:
> 
> * release insights-admission 1.0.0
> * release insights-agent 2.0.0 (with insights-admission 1.0.0 as a dependency)
> 
> To do that, you can create a new branch (say, `bm/admission-1-0`) and run
> 
> ```
> git checkout bm/agent-2-0 ./stable/insights-admission
> ```

noted let me do that ..

**Comment by ivanfetch-wt** [comment]
I'm aware that when I squash-merged a PR into this branch, via [this commit](https://github.com/FairwindsOps/charts/pull/697/commits/3fb0d90d73c0865f0af1d75373176d7b18268391) I referenced an incorrect insights-plugins PR in that commit note. I don't think this is worth reverting to fix, but wanted to comment here in case someone later finds my commit msg confusing.

File: stable/insights-agent/ci/test-values.yaml [hunk 3]
```diff
@@ -78,9 +71,12 @@ trivy:
     requests:
       cpu: 50m
 
-resourcemetrics:
+prometheus-metrics:
   enabled: true
-  installPrometheus: true
+  installPrometheusServer: true
```
Comment (sudermanjr) [inline stable/insights-agent/ci/test-values.yaml:76]: This hasn't been changed in the requirements.yaml for the subchart, which is causing prometheus to get added unexpectedly

**Review by sudermanjr — CHANGES_REQUESTED** [review]

File: stable/insights-admission/Chart.yaml [hunk 4]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: insights-admission
 description: A Validating Webhook Admission Controller that utilizes Fairwinds Insights.
 type: application
-version: 1.0.3
+version: 1.1.0
```
Comment (sudermanjr) [inline stable/insights-admission/Chart.yaml:5]: If we go ahead and bump this to 2.0.0, the CI tests will stop trying to do the upgrade check.

**Review by sudermanjr — DISMISSED** [review]

File: stable/insights-agent/values.yaml [hunk 5]
```diff
@@ -280,9 +298,9 @@ pluto:
       cpu: 100m
       memory: 128Mi
 
-resourcemetrics:
+prometheus-metrics:
```
Comment (rbren) [inline stable/insights-agent/values.yaml:301]: I think we need to change this tag to 1.0 now

File: stable/insights-agent/README.md [hunk 6]
```diff
@@ -83,34 +82,37 @@ Parameter | Description | Default
...
 `goldilocks.controller.flags.exclude-namespaces` | Namespaces to exclude from the goldilocks report | `kube-system`
 `goldilocks.vpa.enabled` | Install the Vertical Pod Autoscaler as part of the Goldilocks installation | true
 `goldilocks.controller.flags.on-by-default` | Goldilocks will by default monitor all namespaces that aren't excluded | true
 `goldilocks.controller.resources` | CPU/memory requests and limits for the Goldilcoks controller |
 `goldilocks.dashboard.enabled` | Installs the Goldilocks Dashboard | false
-`resourcemetrics.installPrometheus` | Install a new Prometheus instance for the resourcemetrics report | false
-`resourcemetrics.address` | The address of an existing Prometheus instance to query in the form `<scheme>://<service-name>.<namespace>[:<port>]` for example `http://prometheus-server.prometheus` | `"http://prometheus-server"`
+`prometheus-metrics.installPrometheusServer` | Install a new Prometheus server instance for the proemetheus report | false
```
Comment (rbren) [inline stable/insights-agent/README.md]:
```suggestion
`prometheus-metrics.installPrometheusServer` | Install a new Prometheus server instance for the prometheus report | false
```

File: stable/insights-agent/Chart.yaml [hunk 7]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: A Helm chart to run the Fairwinds Insights agent
 name: insights-agent
-version: 1.17.29
+version: 2.0.0-alpha
```
Comment (sudermanjr) [inline stable/insights-agent/Chart.yaml]:
```suggestion
version: 2.0.0-alpha.0
```

**Review by sudermanjr — CHANGES_REQUESTED** [review]

File: stable/insights-agent/Chart.yaml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: A Helm chart to run the Fairwinds Insights agent
 name: insights-agent
-version: 1.17.29
+version: 2.0.0-rc1
```
Comment (sudermanjr) [inline stable/insights-agent/Chart.yaml]: Proper semver

```suggestion
version: 2.0.0-rc.1
```

**Review by sudermanjr — CHANGES_REQUESTED** [review]

**Review by sudermanjr — APPROVED** [review]

**Review by rbren — APPROVED** [review]

**Pull request merged**